azurerm/internal/services/containers/data_source_kubernetes_cluster.go

```diff
@@ -516,6 +560,11 @@ func dataSourceArmKubernetesClusterRead(d *schema.ResourceData, meta interface{}
 			return fmt.Errorf("Error setting `agent_pool_profile`: %+v", err)
 		}
 
+		kubeletIdentity := flattenKubernetesClusterIdentityProfile(props.IdentityProfile)
```

Comment:
since the fields for a Data Source can differ, we intentionally duplicate the flatten functions - as such can we pull this in here?